extensions/amp-a4a/0.1/test/test-real-time-config-manager.js

```diff
@@ -0,0 +1,426 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createElementWithAttributes} from '../../../../src/dom';
+import {isFiniteNumber} from '../../../../src/types';
+import {AmpA4A} from '../amp-a4a';
+import {
+  maybeExecuteRealTimeConfig_,
+  validateRtcConfig_,
+  RTC_ERROR_ENUM,
+} from '../real-time-config-manager';
+import {Xhr} from '../../../../src/service/xhr-impl';
+// Need the following side-effect import because in actual production code,
+// Fast Fetch impls are always loaded via an AmpAd tag, which means AmpAd is
+// always available for them. However, when we test an impl in isolation,
+// AmpAd is not loaded already, so we need to load it separately.
+import '../../../amp-ad/0.1/amp-ad';
+
+describes.realWin('real-time-config-manager', {amp: true}, env => {
+  let element;
+  let a4aElement;
+  let sandbox;
+  let fetchJsonStub;
+
+  beforeEach(() => {
+    sandbox = env.sandbox;
+    env.win.AMP_MODE.test = true;
+    const doc = env.win.document;
+    // TODO(a4a-cam@): This is necessary in the short term, until A4A is
+    // smarter about host document styling.  The issue is that it needs to
+    // inherit the AMP runtime style element in order for shadow DOM-enclosed
+    // elements to behave properly.  So we have to set up a minimal one here.
+    const ampStyle = doc.createElement('style');
+    ampStyle.setAttribute('amp-runtime', 'scratch-fortesting');
+    doc.head.appendChild(ampStyle);
+    element = createElementWithAttributes(env.win.document, 'amp-ad', {
+      'width': '200',
+      'height': '50',
+      'type': 'doubleclick',
+      'layout': 'fixed',
+    });
+    doc.body.appendChild(element);
+    fetchJsonStub = sandbox.stub(Xhr.prototype, 'fetchJson');
+    a4aElement = new AmpA4A(element);
+
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  function setFetchJsonStubBehavior(params, response, isString, shouldFail) {
+    const textFunction = () => {
+      return !isString ? Promise.resolve(JSON.stringify(response)) :
+          Promise.resolve(response);
+    };
+    if (shouldFail) {
+      fetchJsonStub.withArgs(params).returns(Promise.reject());
+    } else {
+      fetchJsonStub.withArgs(params).returns(Promise.resolve({
+        status: 200,
+        text: textFunction,
+      }));
+    }
+  }
+
+  function setRtcConfig(rtcConfig) {
+    element.setAttribute('rtc-config', JSON.stringify(rtcConfig));
+  }
+
+  describe('#maybeExecuteRealTimeConfig_', () => {
+    function executeTest(args) {
+      const rtcConfig = {
+        urls: args.urls,
+        vendors: args.vendors,
+        timeoutMillis: args.timeoutMillis};
+      setRtcConfig(rtcConfig);
```

Comment:
Might as well create the object within setRtcConfig call given you don't use it otherwise